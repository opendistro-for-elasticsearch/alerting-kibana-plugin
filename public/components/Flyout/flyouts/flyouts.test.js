/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import Flyouts from './index';

describe('Flyouts.message', () => {
  test('generates message JSON', () => {
    const json = Flyouts.message();
    expect(json).toMatchSnapshot();
  });
});

describe('Flyouts.messageFrequency', () => {
  test('generates message JSON', () => {
    const json = Flyouts.messageFrequency();
    expect(json).toMatchSnapshot();
  });
});

describe('Flyouts.triggerCondition', () => {
  test('generates message JSON', () => {
    const json = Flyouts.triggerCondition({});
    expect(json).toMatchSnapshot();
  });
});
