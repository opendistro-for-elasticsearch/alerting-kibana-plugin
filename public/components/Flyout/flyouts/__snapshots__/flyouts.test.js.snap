// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flyouts.message generates message JSON 1`] = `
Object {
  "body": <EuiText
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    <p>
      You have access to a "ctx" variable in your painless scripts and action mustache templates.
    </p>
    <h3>
      Learn More
    </h3>
    <ul>
      <li>
        <EuiLink
          color="primary"
          href="https://mustache.github.io/mustache.5.html"
          target="_blank"
          type="button"
        >
          HTML Templates with Mustache.js
        </EuiLink>
      </li>
    </ul>
  </EuiText>,
  "flyoutProps": Object {
    "aria-labelledby": "messageFlyout",
    "maxWidth": 500,
    "size": "m",
  },
  "header": <EuiTitle
    size="m"
    style={
      Object {
        "fontSize": "25px",
      }
    }
  >
    <h2>
      <strong>
        Message
      </strong>
    </h2>
  </EuiTitle>,
  "headerProps": Object {
    "hasBorder": true,
  },
}
`;

exports[`Flyouts.messageFrequency generates message JSON 1`] = `
Object {
  "body": <EuiText
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    <p>
      Specify message frequency to limit the number of notifications you receive within a given span of time. This setting is especially useful for low severity trigger conditions.
    </p>
    <p>
      Consider the following example:
    </p>
    <ul>
      <li>
        A trigger condition is met.
      </li>
      <li>
        The monitor sends a message
      </li>
      <li>
        Message frequency is set to 10 minutes.
      </li>
    </ul>
    <p>
      For the next 10 minutes, even if a trigger condition is met dozens of times, the monitor sends no additional messages. If the trigger condition is met 11 minutes later, the monitor sends another message.
    </p>
  </EuiText>,
  "flyoutProps": Object {
    "aria-labelledby": "messageFrequencyFlyout",
    "maxWidth": 500,
    "size": "m",
  },
  "header": <EuiTitle
    size="m"
    style={
      Object {
        "fontSize": "25px",
      }
    }
  >
    <h2>
      <strong>
        Message frequency
      </strong>
    </h2>
  </EuiTitle>,
  "headerProps": Object {
    "hasBorder": true,
  },
}
`;

exports[`Flyouts.triggerCondition generates message JSON 1`] = `
Object {
  "body": <div>
    <EuiText
      style={
        Object {
          "fontSize": "14px",
        }
      }
    >
      <p>
        You have access to a "ctx" variable in your painless scripts
      </p>
      <p>
        Below shows a quick JSON example of what's available under the "ctx" variable along with the actual results (where possible) for you to reference.
      </p>
    </EuiText>
    <EuiSpacer
      size="m"
    />
    <EuiCodeBlock
      language="json"
    >
      {
    "monitor": "...",
    "trigger": "...",
    "results": "...",
    "periodStart": "...",
    "periodEnd": "...",
    "alert": "...",
    "error": "..."
}
    </EuiCodeBlock>
    <EuiSpacer
      size="m"
    />
    <EuiCodeEditor
      height="700px"
      mode="json"
      readOnly={true}
      setOptions={
        Object {
          "fontSize": "12px",
        }
      }
      theme="github"
      value="{}"
      width="100%"
    />
  </div>,
  "flyoutProps": Object {
    "aria-labelledby": "triggerConditionFlyout",
    "maxWidth": 500,
    "size": "m",
  },
  "header": <EuiTitle
    size="m"
    style={
      Object {
        "fontSize": "25px",
      }
    }
  >
    <h2>
      <strong>
        Trigger condition
      </strong>
    </h2>
  </EuiTitle>,
  "headerProps": Object {
    "hasBorder": true,
  },
}
`;
